cartoframes/dataset.py

```diff
@@ -376,14 +382,81 @@ def get_columns(context, query):
 
 def get_query(dataset):
     if isinstance(dataset, Dataset):
-        return dataset.query or _default_query(dataset)
+        if dataset.query is None:
+            return _compute_query(dataset)
+        else:
+            return dataset.query
 
 
-def _default_query(dataset):
+def _compute_query(dataset):
     if dataset.table_name and dataset.schema:
         return 'SELECT * FROM "{0}"."{1}"'.format(dataset.schema, dataset.table_name)
 
 
+def get_geodataframe(dataset):
+    if isinstance(dataset, Dataset):
+        if dataset.gdf is None:
+            return _compute_geodataframe(dataset)
+        else:
+            return dataset.gdf
+
+
+def _compute_geodataframe(dataset):
+    if dataset.df is not None:
+        df = dataset.df.copy()
+        geom_column = _get_geom_column(df)
+        if geom_column is not None:
+            df['geometry'] = _compute_geometry_from_geom(geom_column)
+        else:
+            lat_column = _get_lat_column(df)
+            lng_column = _get_lng_column(df)
+            if lat_column is not None and lng_column is not None:
+                df['geometry'] = _compute_geometry_from_latlng(lat_column, lng_column)
+            else:
+                raise ValueError('DataFrame has no geographic data.')
+        return geopandas.GeoDataFrame(df)
+
+
+def _get_geom_column(df):
+    if 'geometry' in df:
+        return df['geometry']
+    if 'the_geom' in df:
+        return df['the_geom']
+    if 'wkb_geometry' in df:
+        return df['wkb_geometry']
+    if 'geom' in df:
+        return df['geom']
+    if 'geojson' in df:
```

Comment:
About geojson, if we support them, we should decode them properly right? I guess decode_geom does not support geojson.

Reply:
I supposed that this was an alias for `geometry` (https://github.com/CartoDB/cartodb/blob/master/services/importer/lib/importer/georeferencer.rb#L14).
IMO we can remove this case since we have good support for geojson in CF

Reply:
let's remove it then.